pkg/client/typed/dynamic/client.go

```diff
@@ -91,6 +98,13 @@ type ResourceClient struct {
 	ns       string
 }
 
+// SubresourceClient is an API interface to a specific subresource of a resource under a
+// dynamic client.
+type SubresourceClient struct {
```

Comment:
I think it's fine as a struct, clients can define an interface if they need one. /channeling @krousey